### ISSUE TITLE
why my 'workers=4,use_multiprocessing=True' do not work while training?

### ISSUE THREAD

**Issue by renguote** [keras-team/keras#17118]
You can obtain the TensorFlow version with:
TF  2.10
**Describe the problem**.
**Describe the current behavior**.
history = model.fit(train_model_input, epochs=1, verbose=1, validation_split=0.0,steps_per_epoch= math.ceil( train_row_len/256), workers=4,use_multiprocessing=True).
As you see, 'workers=4,use_multiprocessing=True'  don't  work while training.I find it only 100% CPU-Util not the expect 400%.
**Describe the expected behavior**.
I can  increase the utilization of my CPU to 400%.

**Standalone code to reproduce the issue**.

import tensorflow as tf
from tensorflow.keras import datasets, layers, models
from tensorflow.keras.utils import Sequence
from math import ceil
import numpy as np
import numpy as np
import math
import os

class DataGenerator(Sequence):
 
    def __init__(self, x_set, y_set, batch_size):
        self.x, self.y = x_set, y_set
        self.batch_size = batch_size

    def __len__(self):
        return ceil(len(self.x) / self.batch_size)

    def __getitem__(self, idx):
        end = min(self.x.shape[0], (idx + 1)*batch_size)
        return self.x[idx*batch_size:end], self.y[idx*batch_size:end]

batch_size = 32
epochs = 100

(train_images, train_labels), (test_images, test_labels) = datasets.cifar10.load_data()

# Normalize pixel values to be between 0 and 1
train_images, test_images = train_images / 255.0, test_images / 255.0
training_generator = DataGenerator(train_images, train_labels, batch_size)

model = models.Sequential()
model.add(layers.Conv2D(320, (3, 3), activation='relu', input_shape=(32, 32, 3)))
model.add(layers.MaxPooling2D((2, 2)))
model.add(layers.Conv2D(64, (3, 3), activation='relu'))
model.add(layers.MaxPooling2D((2, 2)))
model.add(layers.Conv2D(64, (3, 3), activation='relu'))
model.add(layers.Flatten())
model.add(layers.Dense(64, activation='relu'))
model.add(layers.Dense(10))

model.compile(optimizer='adam',
              loss=tf.keras.losses.SparseCategoricalCrossentropy(from_logits=True),
              metrics=['accuracy'])

history = model.fit(x=training_generator,
                    steps_per_epoch=train_images.shape[0]//batch_size,
                    validation_data=(train_images, train_labels),
                    epochs=epochs,
                    verbose=1,
                    use_multiprocessing=True,
                    workers=4)

**Source code / logs**.

**Comment by tilakrayal** [comment]
@renguote,
Code shared is full of [indentation errors](https://colab.research.google.com/gist/tilakrayal/6a44ac588014efda771584bc815567c8/untitled659.ipynb), please share a colab gist with issue reported or simple stand alone indented code with all dependencies such that we can replicate the issue reported.

**Comment by renguote** [comment]
[The code](https://colab.research.google.com/gist/renguote/2128f3d0f633abbb52083d788add6bae/untitled659.ipynb) already works correctly，Thank you!

**Comment by renguote** [comment]
![image](https://user-images.githubusercontent.com/114635552/195012826-06299568-67c2-4162-8f12-773d3cbc8d10.png)

I start 4 processes, but only one process is working, the others are not working.

**Comment by tilakrayal** [comment]
@renguote,
I tried to execute the given code and the execution runtime was more than expected and gist attached [here](https://colab.research.google.com/gist/tilakrayal/e5d2dadbc5c11d3f5e0ef6bff7d983c6/untitled675.ipynb). Could you please try to reduce the no. of epochs and the batch size and try to test on new virtual environment. Thank you!

**Comment by renguote** [comment]
[The code](https://colab.research.google.com/gist/renguote/beb493d684e5ff6dc0a2d81157adbcbb/untitled675.ipynb)  already works correctly. My problem is when I start 4 processes in keras(model.fit(..., use_multiprocessing=True,workers=4)),  but only one process works, Thank you!

**Comment by PierrickPochelu** [comment]
The `top` and your explanation show 5 processes: 1 training process and 4 loading processes.  The forward/backward computing (training process) is bottlenecking the program speed, and not the loading of batches. I think the majority of deep learning applications are in this case.

**Comment by renguote** [comment]
Thank you!

**Comment by tilakrayal** [comment]
@renguote,
Could you please confirm if the issue is resolved. if yes, please feel free to move this issue to closed status. Thank you!
